### PR TITLE
Feature/remaining war attacks part 3

### LINKED ISSUE #52: Feature: API to check remaining war attacks on current war
### Notes on Response body

Can use the following endpoint to determine the clans that are participating in the riverwar respective to the clan tag as the input:
```
https://api.clashroyale.com/v1/clans/<clan_tag>/currentriverrace

example: where <clan_tag> is %239GULPJ9L for ausclan
```
Note: 

- endpoint is from official Royale API
- grab the clans item from the response
- participants for each clan could be greater than 50 because people come and go

### How to get remainingPlayer
- out of the participants, grab all who has decksUsedToday != 0, and then count up deckUsedToday != 4.
- from the result, do 50 - the result to find out the # of participants who hasnt done their war.

Note: != 0 can tell us how many minis we can bring into the clan

### Output requirements

- A list of Clans (5 max) - ordered by Medals (highest to lowest)
- Right before table output, explain what each column represents (follow similar format/bold/spacing as other command outputs)
```
  Clan        | Medals | Participated | DecksRemaining  | PlayersRemaining |
--------------+--------+--------------+-----------------+------------------|
  Ausclan     | 29860  |    44/50     | 10 decks        |  6 players       |
  ifTrashClan | 28000  |    40/50     | 30 decks        | 10 players       |
  soClanYou   | 26400  |    30/50     | 70 decks        | 20 players       |
  otherClan   |  5000  |    50/50     | 150 decks       | 50 players       |
```
Note: 
- can call the endpoint `/clans/{clanTag}` to get list of current 50 players in clan
- **Medals** corresponds to `periodPoints` in the API response. It is the number of medals won in current war day (NOT the total movement points over the week)
- **PlayersRemaining** : players who haven't used all 4 attacks AND are currently in clan (deckUsedToday != 4 and CURRENTLY in clan)
- **DecksRemaining**: the amount of decks that hasn't been used:
    - `(50 - num_players_used_deck_left_clan) * 4 - sum_of("deckUsedToday for all CURRENT clan members")`
- **Participated**: players who used any attacks even if they're no longer in clan this week ( deckUsedToday != 0) 

### Steps:
1. Ingest clash royale currentriverrace and clan API and format it into appropriate data structure
2. Transform the ingested data into the summary table above
3. Display table into discord bot

### PR DESCRIPTION
**Pull request by josephwccheng**
Closes #52

### CODE REVIEW

File: cogs/clash_river_war.py [hunk 1]
```diff
@@ -111,6 +111,33 @@ def clan_river_race_history_embed(clan_info: Dict, clan_players_war_history: Dic
...
+    )
+
+    columns = '**Clan**|**Medals**|**Fame**|**Participated**|**DecksRemaining**|**PlayersRemaining**'
+    row_val = []
+    for clan_attacks in all_clan_attacks:
+        row_val.append('`{:^15}`|`{:^5}`|`{:^6}`|`{:^6}`|`{:^9}`|`{:^10}`'.format(
+            clan_attacks.name,
+            clan_attacks.medals,
+            clan_attacks.fame,
+            str(clan_attacks.participated) + '/50',
```
Comment (damoster) [inline cogs/clash_river_war.py:134]: /nit yuck str(), us f-strings, they're the future. i.e. instead of
```python
str(clan_attacks.participated) + '/50'
```
Do this
```python
f'{clan_attacks.participated}/50'
```

File: cogs/clash_river_war.py [hunk 2]
```diff
@@ -111,6 +111,33 @@ def clan_river_race_history_embed(clan_info: Dict, clan_players_war_history: Dic
...
+            **DecksRemaining** - How many decks in the clan remaining that can still be used for the Current River Race
+            **PlayersRemaining** - How many players can still do the war, ignoring those who left and those who participated already
+        '''.format()),
+        colour=discord.Colour.green()
+    )
+
+    columns = '**Clan**|**Medals**|**Fame**|**Participated**|**DecksRemaining**|**PlayersRemaining**'
+    row_val = []
+    for clan_attacks in all_clan_attacks:
+        row_val.append('`{:^15}`|`{:^5}`|`{:^6}`|`{:^6}`|`{:^9}`|`{:^10}`'.format(
```
Comment (damoster) [inline cogs/clash_river_war.py:130]: I just had an idea, to cater for varying clan names, we could change the forced column min width to match that of the max length of longest clan name of the list. i.e. instead of
```python
for clan_attacks in all_clan_attacks:
    row_val.append('`{:^15}`|`{:^5}`|`{:^6}`|`{:^6}`|`{:^9}`|`{:^10}`'.format(
        clan_attacks.name,
        clan_attacks.medals,
        #...
```
Do this
```
max_name_len = max([len(c.name) for c in all_clan_attacks])
for clan_attacks in all_clan_attacks:
    row_val.append('`{}`|`{:^5}`|`{:^6}`|`{:^6}`|`{:^9}`|`{:^10}`'.format(
        f"{c.name}:{^{max_name_len}}",
        clan_attacks.medals,
        #...
```

Comment (damoster) [reply cogs/clash_river_war.py:130]: Might be worth doing this for all the other discord commands where name is NOT the last column

File: discord_bot.py [hunk 3]
```diff
@@ -41,7 +41,7 @@ async def on_command_error(self, ctx, exc: Exception):
             message = "Command used is missing required argument"
             logging.warn(get_stack_trace_str(exc))
         else:
-            message = "Oh no! Something went wrong while running the command!"
+            message = "Oh no! Something went wrong while running the command! Please use **!bothelp** command for supported commands."
```
Comment (damoster) [inline discord_bot.py:44]: Why was this added? This message is posted when something unexpected happens (i.e. bug in our program). Asking bothelp won't help lol

**Review by damoster — CHANGES_REQUESTED** [review]
Mostly looks good. Just some small things to address

File: cogs/clash_river_war.py (hunk 2 above)
Comment (josephwccheng) [reply cogs/clash_river_war.py:130]: done!

File: discord_bot.py (hunk 3 above)
Comment (josephwccheng) [reply discord_bot.py:44]: explained with david - 4xx, 5xx error code differentiation

**Review by damoster — APPROVED** [review]
Looks good. Merge away
